astroid/nodes/node_classes.py

```diff
@@ -4489,12 +4489,12 @@ class MatchSingleton(Pattern):
     def __init__(
         self,
         *,
-        value: Literal[True, False, None],
+        value: Optional[bool],
```

Comment:
The `Literal` doesn't work when the `init` is called via `super()` when that inheritor does use `bool`. I don't think `Literal` provides any added benefit here.

Reply:
This should really stay as `Literal`. A PR recently should have added support for it. Just don't know if it's already included in the next release or the one after that. Please add `type: ignore` if necessary.
https://github.com/python/mypy/pull/10389

Reply:
Didn't know there is a difference between the two.

I'll add a ignore.